mmv1/templates/terraform/encoders/compute_backend_bucket.go.erb

```diff
@@ -0,0 +1,35 @@
+c, cdnPolicyOk := d.GetOk("cdn_policy")
+
+if !cdnPolicyOk {
+  return obj, nil
+}
+
+cdnPolicies := c.([]interface{})
+
+if len(cdnPolicies) == 0 {
+  return obj, nil
+}
+
+cdnPolicy := cdnPolicies[0].(map[string]interface{})
+
+cacheMode := cdnPolicy["cache_mode"].(string)
+_, defaultTTLOk := cdnPolicy["default_ttl"]
+_, clientTTLOk := cdnPolicy["client_ttl"]
+
+if obj["cdnPolicy"] == nil {
+  obj["cdnPolicy"] = make(map[string]interface{})
+}
+
+encCDNPolicy := obj["cdnPolicy"].(map[string]interface{})
+
+switch cacheMode {
+case "USE_ORIGIN_HEADERS":
+  if _, ok := encCDNPolicy["defaultTtl"]; ok && defaultTTLOk {
```

Comment:
note should we throw an error if v is not 0? That would infer that the user is trying to set the value even though it's unsettable with `USE_ORIGIN_HEADERS`

```golang
v, ok := encCDNPolicy["defaultTtl"]
if ok && v!= 0 {
// error?
}
```